Packs/Core/Scripts/SearchIndicatorInEvents/SearchIndicatorInEvents.yml

```diff
@@ -0,0 +1,48 @@
+commonfields:
+  id: SearchIndicatorInEvents
+  version: -1
+name: SearchIndicatorInEvents
+script: ''
+type: python
+tags:
+  - Utilities
+comment: Searches for a specific indicator in the tenant's events and logs data and extract the logs which the indicator appears in.
+enabled: true
+args:
+  - name: indicator
+    required: true
+    description: The indicator value (e.g., IP address, domain, hash) to search for in the selected dataset logs.
+  - name: time_frame
+    required: true
+    defaultValue: 7 days
+    description: The time range to search within, specified in days (e.g., "7 days" means searching the past 7 days of data).
+  - name: data_set
+    defaultValue: xdr_data
+    description: The name of the dataset to search in. If not specified, the search defaults to the "xdr_data" dataset.
+  - name: query_name
+    required: true
+    description: A user-defined name. The query results will appear under the query_name in the context data of the current war room.
+  - name: query_id
+    hidden: true
+    description: Internal Id of the query, hidden from inserting manually.
+  - name: interval_in_seconds
+    defaultValue: '30'
+    description: The interval, in seconds, between checking the status of the query while waiting for the query to complete.
+  - name: timeout_in_seconds
+    defaultValue: '600'
+    description: The maximum time, in seconds, to wait for the query to finish before the command is forced to fail.
+outputs:
+  - contextPath: PaloAltoNetworksXQL
+    description: A list of event records (constructed as dictionaries) where the specified indicator was found during the search.
```

Comment:
A list of event records (constructed as dictionaries) where the specified indicator was found.